src/iOS.Core/Controllers/LoginPasswordViewController.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using UIKit;
+using Foundation;
+using Bit.iOS.Core.Views;
+using Bit.App.Resources;
+using Bit.iOS.Core.Utilities;
+using Bit.App.Abstractions;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using System.Threading.Tasks;
+using Bit.Core.Models.Domain;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace Bit.iOS.Core.Controllers
+{
+    public abstract class LoginPasswordViewController : ExtendedUITableViewController
+    {
+        private IDeviceActionService _deviceActionService;
+        private IStorageService _storageService;
+        private IAuthService _authService;
+        private ISyncService _syncService;
+
+        public LoginPasswordViewController(IntPtr handle)
+            : base(handle)
+        { }
+
+        public abstract UINavigationItem BaseNavItem { get; }
+        public abstract UIBarButtonItem BaseCancelButton { get; }
+        public abstract UIBarButtonItem BaseSubmitButton { get; }
+        public abstract Action Success { get; }
+        public abstract Action Cancel { get; }
+
+        public FormEntryTableViewCell EmailCell { get; set; } = new FormEntryTableViewCell(
+            AppResources.EmailAddress);
+        public FormEntryTableViewCell MasterPasswordCell { get; set; } = new FormEntryTableViewCell(
+            AppResources.MasterPassword);
+
+        public override void ViewDidLoad()
+        {
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>("deviceActionService");
+            _storageService = ServiceContainer.Resolve<IStorageService>("storageService");
+            _authService = ServiceContainer.Resolve<IAuthService>("authService");
+            _syncService = ServiceContainer.Resolve<ISyncService>("syncService");
+
+            BaseNavItem.Title = AppResources.Bitwarden;
+            BaseCancelButton.Title = AppResources.Cancel;
+            BaseSubmitButton.Title = AppResources.LogIn;
+
+            var descriptor = UIFontDescriptor.PreferredBody;
+            
+            MasterPasswordCell.TextField.SecureTextEntry = true;
+            MasterPasswordCell.TextField.ReturnKeyType = UIReturnKeyType.Go;
+            MasterPasswordCell.TextField.ShouldReturn += (UITextField tf) =>
+            {
+                // TODO This isn't working - bricks application
+                LogInAsync().GetAwaiter().GetResult();
+                return true;
+            };
+
+            TableView.RowHeight = UITableView.AutomaticDimension;
+            TableView.EstimatedRowHeight = 70; // Should this be adjusted for two cells?
+            TableView.Source = new TableSource(this);
+            TableView.AllowsSelection = true;
+            
+            // Attempt to programatically set the user's email
+            EmailCell.TextField.Text = _storageService.GetAsync<string>("rememberedEmail").GetAwaiter().GetResult();
```

Comment:
This is another example of the pattern above, however `IStorageService` implementation here may only be going off of in-memory, but another implementation that actually waits on I/O operations or engages in true async behavior may cause a similar deadlock scenario.

Reply:
Updated - Xamarin provides asnyc overrides for lifecycle methods.